### ISSUE TITLE
Archived tables not detached from parent and dropped from publication, Postgresql v14.8

### ISSUE THREAD

**Issue by rsquaretrade** [pgpartman/pg_partman#577]
Hello, when we first upgraded from PG10 to PG14 using table inheritance partitioning, we noticed that archived tables were not being dropped from publication, as a workaround, we just do a "NO INHERIT" whenever the table breaks replication every month.  

We are now migrating from table inheritance to native partitioning and during testing I noticed the same behavior--archived tables are not detached from the parent table and not dropped from publication.  Is there a setting in partman.part_config to detach and drop from pub?  Thanks in advance.

SELECT * FROM pg_publication_tables where pg_publication_tables.pubname='x_pub';
┌──────────┬──────────────┬──────────────────────┐
 │ pubname                      │ schemaname   │   tablename          │
├──────────┼──────────────┼──────────────────────┤
│ mmdm_pub.                 │ dba                     │           x_p2024_01 │
│ mmdm_pub                  │ dba                     │           x_p2023_09 │
│ mmdm_pub                  │ dba                     │           x_p2023_10 │
│ mmdm_pub                  │ dba                     │           x_p2023_11 │
│ mmdm_pub                  │ dba                     │           x_p2023_12 │
│ mmdm_pub                  │ dba                     │           x_default  │
│ mmdm_pub                  │ dba_archive       │           x_p2023_04 │
└──────────┴──────────────┴──────────────────────

I did some more testing, OLTP:  retention=3 months, OLAP retention=24 months.  Since the tables are now lop-sided (there will be more tables in DW), I was concerned that I will get a replication error.  I did run_maintenance in OLTP and it archived fine.  I did run_maintenance in OLAP and it did NOT archive as expected due to a longer retention but I also did not get a replication issue.  Which is great but I don't quite understand.  Is it bec the Publication  "knows" retention =24 months in the Subscriber?

**Comment by keithf4** [comment]
What are the contents of your `part_config` table for this partition set?

**Comment by rsquaretrade** [comment]
Thanks in advance.  More importantly, since pg_partman in production is still in version 4.4, we want to know if there's a feature in pg_partman to auto detach/disinherit tables moving to mmd_archive as it is currently causing error in the logs: 
 "mmdm_archive does not exist".

**Comment by keithf4** [comment]
So the retention system should be detaching any tables older than the intervals given in the `retention` column. 

So anything older than 4 months on publisher and anything older than 24 months on subscriber side. And since you have `retention_schema` set, it should be moving the child tables to the designated schema. Going by the SELECT statement from the publisher table you gave, I'm not seeing any tables older than 4 months listed in the original schema except one older one in the `dba_archive` schema from April? However you have the retention schema set to `x_archive`.  So really not sure what is going on there.

The publisher/subscriber is completely unaware of any retention being done as far as partman is concerned. The only thing that the subscriber cares about is that any tables that exist on the publisher side must exist on the subscriber side. How this affects a partitioned table being published/subscribed I'm not sure. For PG14 and above, child tables in a declarative partition set are now automatically added to the publication if the parent table was added to the publication, so you don't have to have pg_partman do that for you anymore (I've actually dropped the feature in the next major version of partman coming out soon). What this means for the subscriber side, I'm actually not quite sure anymore. They may need to 100% match, but I wouldn't be able to say without experimenting. I'm not sure when I'll get to do that at the moment.

I don't think I have partman set to remove a table from a publication if it's been detached as part of retention
